caddy/caddymain/run.go

```diff
@@ -19,6 +19,8 @@ import (
 	"github.com/mholt/caddy"
 	// plug in the HTTP server type
 	_ "github.com/mholt/caddy/caddyhttp"
+	// plugin the DNS server type
+	_ "github.com/miekg/caddy-coredns/coredns"
```

Comment:
Yep, this is exactly how easy it is to add the DNS server type!

However we probably won't ship the DNS server _by default_ (HTTP will remain the default). I've already updated the Download page so DNS can be checked very easily just like other features. So we can remove this line and people can still get the DNS server quite easily. Hope that's okay.